### DISCUSSION TITLE
Nearest Neighbor for square grid KAM

### DISCUSSION THREAD

**Discussion by lkuna24** [mtex-toolbox/mtex#1904] [Ask Anything] [answered]
According to the documentation for computing KAM, it seems like a Von Neumann neighborhood is utilized.  Is it possible to select a Moore neighborhood for evaluation of KAM on a square grid?  (https://en.wikipedia.org/wiki/Moore_neighborhood).  Or is this a feature that would need to be implemented?

**Comment by ralfHielscher** [comment, marked as answer]
The following will work for a square grid
```matlab
ebsd = ebsd.gridify
n = 2; % number of neighbours
plot(ebsd, ebsd.KAM('weights',ones(2*n+1)))
```
Here you use the option `weights` to pass a filter to the KAM function which describes the neighbourhod.

Ralf.

**Reply by lkuna24** [reply]
Thank you Ralf.
